nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HealthCheckWithThresholdExecutionOptions.java

```diff
@@ -53,9 +55,30 @@ public class HealthCheckWithThresholdExecutionOptions extends HealthCheckExecuti
     public HealthCheckWithThresholdExecutionOptions(boolean enabled, long time, TimeUnit unit, boolean addToMicroProfileHealth, String
             thresholdCritical, String thresholdWarning, String thresholdGood) {
         super(enabled, time, unit, addToMicroProfileHealth);
-        this.thresholdCritical = Integer.parseInt(thresholdCritical);
-        this.thresholdWarning = Integer.parseInt(thresholdWarning);
-        this.thresholdGood = Integer.parseInt(thresholdGood);
+
+        try {
+            this.thresholdCritical = Integer.parseInt(thresholdCritical);
+        }
+        catch (Exception e) {
+            this.thresholdCritical = HealthCheckConstants.THRESHOLD_DEFAULTVAL_CRITICAL_INT;
+            _logger.warning("Could not parse critical threshold: " + thresholdCritical);
```

Comment:
This is saying that it can't parse the default - you need to log what it couldn't parse before setting the variable to the default.

It's also probably worth mentioning that the default value will be used because of the failure to parse the configured value.